### DISCUSSION TITLE
Code snippets [Sep 17]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#3806] [Discord Threads]
> 🙋‍♂️ **emdax**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### On v0.96.2, you can find sample `code snippets` menu here  
![image0](https://cdn.discordapp.com/attachments/1018070639304781864/1018077472983629835/Screenshot_2022-09-10_at_2.05.15_PM.png)  
#  
> 👨‍🦲 **dstala**  
#### We are updating documents as per new UI. should be ready in few days  
#  
> 👨‍🦲 **dstala**  
#### this menu will open-up code snippets for your project table in in different programming languages  
#  
> 🙋‍♂️ **emdax**  
#### thank you  
#  
> 🙋‍♂️ **emdax**  
#### Does the API snippet update to reflect Filter/Sort setting of view?  
#  
> 👨‍🦲 **dstala**  
#### No. Snippet is static. Something like below for JS   
####    
#### ```shell   
#### import axios from "axios";   
####    
#### const options = {   
####   method: 'GET',   
####   url: 'http://localhost:8080/api/v1/db/data/noco/p_6bba84fwrupxcj/City/views/City',   
####   params: {offset: '0', limit: '25', where: ''},   
####   headers: {   
####     'xc-auth': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InVzZXJAbm9jb2RiLmNvbSIsImZpcnN0bmFtZSI6bnVsbCwibGFzdG5hbWUiOm51bGwsImlkIjoidXNfeHBrZm12dzBmM3oyZHQiLCJyb2xlcyI6InVzZXIsc3VwZXIiLCJ0b2tlbl92ZXJzaW9uIjoiZDdmNTI2YzcwM2JmNmM0N2VjZjNjYjUyZmRjMjUxYjU4NWJiYTFiMDVjMjAyY2IyNjVjYTQ5NDNhZGZlNzMyMmU1ZTlhYjA2MWM1MTk5Y2IiLCJpYXQiOjE2NjI3OTg4ODUsImV4cCI6MTY2MjgzNDg4NX0.sB3pBy7_IAu4LvcrZ-AJ29OhKd-VbFiLzvy1u-wHXcM'   
####   }   
#### };   
####    
#### axios.request(options).then(function (response) {   
####   console.log(response.data);   
#### }).catch(function (error) {   
####   console.error(error);   
#### });   
#### ```  
#  
> 🙋‍♂️ **emdax**  
#### getting:  Request header field xc-auth is not allowed by Access-Control-Allow-Headers in preflight response.  
#  
> 👨‍🦲 **dstala**  
#### You can also try Swagger for experimenting with APIs  
![image0](https://cdn.discordapp.com/attachments/1018070639304781864/1018080938724446218/Screenshot_2022-09-10_at_2.19.41_PM.png)  
#  
> 🙋‍♂️ **emdax**  
#### Error: Request header field xc-auth is not allowed by Access-Control-Allow-Headers in preflight response.  
#  
> 👨‍🦲 **dstala**  
#### <@372124593848057866> can you provide details about the API that you are trying out? with any sensitive information redacted?  
#  
> 🙋‍♂️ **emdax**  
#### I just want to load filtered sorted view  
#  
> 🙋‍♂️ **emdax**  
#### in JSON  
#  
> 🙋‍♂️ **emdax**  
#### with fetch  
#  
> 💁 **o1lab**  
#### you would have to use API token with xc-token header  
#  
> 💁 **o1lab**  
#### https://docs.nocodb.com/developer-resources/accessing-apis#api-token  
#  
> 💁 **o1lab**  
#### xc-auth has a default expiry of 1 day  
#  
> 🙋‍♂️ **emdax**  
#### How do you set up NocoDB , to run on PORT 80 with SSL/HTTPS  ?  
#  
> 🙋‍♂️ **emdax**  
#### Use NGINX ?  
#  
> 🙋‍♂️ **emdax**  
#### API is giving CORS errors when called from the browser  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1018070639304781864)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
